javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -61,6 +62,20 @@
         return results;
     }
 
+    @SuppressWarnings("unchecked")
+    public static <K, V, K2, U extends Map<K2, V>> U mapKeys(Map<K, V> source, U zero, Function<? super K, ? extends K2> keyMapper, Function2<V, V, V> valueMerge) {
+        Objects.requireNonNull(source, "source is null");
+        Objects.requireNonNull(zero, "zero is null");
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        Objects.requireNonNull(valueMerge, "valueMerge is null");
+        return source.foldLeft(zero, (acc, entry) -> {
+            K2 k2 = keyMapper.apply(entry._1());
```

Comment:
In Javaslang it is best practice to make local vars `final` if possible to emphasis the _value of values_.